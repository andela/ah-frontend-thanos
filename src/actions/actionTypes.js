const actionTypes = {
  GET_USER_INPUT: 'GET_USER_INPUT',
  USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
  USER_REGISTER_FAIL: 'USER_REGISTER_FAIL',
};

export default actionTypes;
