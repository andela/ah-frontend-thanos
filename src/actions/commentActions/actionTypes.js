
const action = {
  GET_ALL_COMMENTS_SUCCESSFUL: 'GET_ALL_COMMENTS_SUCCESSFUL',
  GET_ALL_COMMENTS_FAIL: 'GET_ALL_COMMENTS_FAIL',
  NEW_COMMENT_DATA: 'NEW_COMMENT_DATA',
  POST_COMMENT_SUCCESSFUL: 'POST_COMMENT_SUCCESSFUL',
  POST_COMMENT_FAIL: 'POST_COMMENT_FAIL',
  LIKEDISLIKEFAIL: 'LIKEDISLIKEFAIL',
  LIKEDISLIKESUCCESSFUL: 'LIKEDISLIKESUCCESSFUL',
};

export default action;
