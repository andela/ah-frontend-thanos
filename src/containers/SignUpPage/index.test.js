import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import SignUpPageConnected, { mapDispatchToProps, SignUpPage } from './index';
import actionTypes from '../../actions/actionTypes';

describe('<SignUpPage />', () => {
  let signUpPageComponent;

  beforeEach(() => {
    const mockStore = configureStore([thunk]);
    const initialState = { articleReducer: { article: {} }, userReducer: { freshUser: {} } };
    signUpPageComponent = shallow(<SignUpPageConnected store={mockStore(initialState)} />);
  });

  it('should render the component', () => {
    expect(signUpPageComponent).toMatchSnapshot();
  });

  it('should dispatch a method to get user input', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getUserInputs({});
    expect(dispatch.mock.calls[0][0]).toEqual({ payload: {}, type: actionTypes.GET_USER_INPUT });
    mapDispatchToProps(dispatch).signUpuser({});
  });

  it('should call signup user method', () => {
    const signUpuser = jest.fn();
    const wrapper = mount(
      <SignUpPage signUpuser={signUpuser} freshUser={{}} getUserInputs={jest.fn()} />,
    );
    wrapper.find('form').simulate('submit');
    expect(signUpuser).toHaveBeenCalled();
  });

  it('should call handle input user method', () => {
    const getUserInputs = jest.fn();
    const wrapper = mount(
      <SignUpPage signUpuser={jest.fn()} freshUser={{}} getUserInputs={getUserInputs} />,
    );
    wrapper.find('input#username').simulate('change');
    expect(getUserInputs).toHaveBeenCalled();
  });
});
