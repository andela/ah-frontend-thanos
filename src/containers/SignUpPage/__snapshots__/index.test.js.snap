// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage /> should render the component 1`] = `
<SignUpPage
  freshUser={
    Object {
      "email": "",
      "password": "",
      "username": "",
    }
  }
  getUserInputs={[Function]}
  signUpuser={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
